frontends/infinite-corridor/src/pages/DemoPage.test.tsx

```diff
@@ -0,0 +1,53 @@
+import { screen, renderTestApp, setMockResponse, user } from "../test-utils"
+import { buildSearchQuery } from "@mitodl/course-search-utils"
+import { makeRequest } from "../test-utils/mockAxios"
+import { assertInstanceOf } from "ol-util"
+
+const getSearchTextInput = (): HTMLInputElement => {
+  const textInput = screen.getByLabelText("Search for")
+  assertInstanceOf(textInput, HTMLInputElement)
+  return textInput
+}
+
+describe("DemoPage", () => {
+  test("the user should be able to update the search text and submit", async () => {
+    setMockResponse.post("search/", {
+      hits: { hits: [], total: 0 }
+    })
+    await renderTestApp({ url: "/demo" })
+
+    const textInput = getSearchTextInput()
+    await user.type(textInput, "New Search Text{Enter}")
+
+    expect(makeRequest.mock.calls[0][0]).toEqual("post")
+    expect(makeRequest.mock.calls[0][1]).toEqual("search/")
+    expect(makeRequest.mock.calls[0][2]).toMatchObject(
+      buildSearchQuery({
+        text:         "New Search Text",
+        from:         0,
+        activeFacets: { type: ["course"] },
+        size:         5
+      })
+    )
```

Comment:
You can also spy on `axios.post` now if you prefer:
```ts
import axios from "../../lib/axios"
// ... then later:
expect(axios.post).toHaveBeenNthCalledWith(0, "/search", body)
```
(*Now you can; last summer this did not work.*)